### PR TITLE
bcdb_mock: make the mock is the same data types as the production bcdb

### LINKED ISSUE #327: bring bcdb mock to same `data` interface as gedis client
Currently we have a bcdb mock we use for development and testing, and a gedis client that actually works against bcdb/gedis. 

Currently both the mock and gedis are growing a part, we need to have similar `data` inputs/outputs from both

### PR DESCRIPTION
**Pull request by zaibon**
fixes #327

### CODE REVIEW

File: cmds/identityd/main.go [hunk 1]
```diff
@@ -95,6 +96,10 @@ func main() {
 		version.ShowAndExit(false)
 	}
 
+	if err := os.MkdirAll(root, 0750); err != nil {
```
Comment (muhamadazmy) [inline cmds/identityd/main.go:99]: identityd failed to start on a clean disk, so this was needed

**Review by muhamadazmy — APPROVED** [review]

File: tools/bcdb_mock/main.go [hunk 2]
```diff
@@ -67,51 +20,67 @@ func main() {
...
+	farmStore, err := LoadfarmStore()
+	if err != nil {
+		log.Fatalln("error loading farm store: %v", err)
+	}
+	resStore, err := LoadProvisionStore()
+	if err != nil {
+		log.Fatalln("error loading provision store: %v", err)
+	}
+
+	// provStore = &provisionStore{Reservations: make([]*reservation, 0, 20)}
```
Comment (zaibon) [inline tools/bcdb_mock/main.go]:
```suggestion
```

File: tools/bcdb_mock/network.go [hunk 3]
```diff
@@ -1,95 +1,11 @@
...
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
-	}
-
-	node.PublicConfig.Type = types.MacVlanIface //TODO: change me once we support other types
-	node.PublicConfig.Version = version + 1
-
-	w.WriteHeader(http.StatusCreated)
-}
+// import (
```
Comment (zaibon) [inline tools/bcdb_mock/network.go]: Let's get rid of this file altogether

File: tools/bcdb_mock/nodes_store.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+		nodeID, ok := mux.Vars(r)[key]
+		if !ok {
+			// programming error, we should panic in this case
+			panic("invalid node-id key")
+		}
+
+		_, err := s.Get(nodeID)
+		if err != nil {
+			// node not found
+			httpError(w, errors.Wrap(err, "node not found"), http.StatusNotFound)
```
Comment (zaibon) [inline tools/bcdb_mock/nodes_store.go]: maybe add the node id in the error message ?

File: tools/bcdb_mock/tnodb_test.go [hunk 5]
```diff
@@ -1,26 +1,26 @@
 package main
 
-import (
-	"fmt"
-	"net"
-	"testing"
+// import (
```
Comment (zaibon) [inline tools/bcdb_mock/tnodb_test.go]: can remove this file too.

**Pull request merged**